enricher/api/src/main/java/io/fabric8/maven/enricher/api/AbstractHealthCheckEnricher.java

```diff
@@ -0,0 +1,90 @@
+package io.fabric8.maven.enricher.api;
+
+import java.util.List;
+
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
+import io.fabric8.kubernetes.api.model.Probe;
+import io.fabric8.kubernetes.api.model.extensions.Deployment;
+
+/**
+ * Enriches containers with health check probes.
+ */
+public abstract class AbstractHealthCheckEnricher extends BaseEnricher {
+
+    public AbstractHealthCheckEnricher(EnricherContext buildContext, String name) {
+        super(buildContext, name);
+    }
+
+    @Override
+    public void addMissingResources(KubernetesListBuilder builder) {
+
+        List<HasMetadata> items = builder.getItems();
+        for (HasMetadata item : items) {
+            if (item instanceof Deployment) {
+                Deployment deployment = (Deployment) item;
+                Container container = getContainer(deployment);
+                if (container != null) {
+                    if (container.getReadinessProbe() == null) {
+                        Probe probe = getReadinessProbe();
+                        if (probe != null) {
+                            log.info("Adding readiness " + describe(probe));
+                            container.setReadinessProbe(probe);
+                        }
+                    }
+
+                    if (container.getLivenessProbe() == null) {
+                        Probe probe = getLivenessProbe();
+                        if (probe != null) {
+                            log.info("Adding liveness " + describe(probe));
+                            container.setLivenessProbe(probe);
+                        }
+                    }
+                }
+            }
+        }
+        builder.withItems(items);
+    }
+
+    private Container getContainer(Deployment deployment) {
+        if (deployment.getSpec() != null &&
+                deployment.getSpec().getTemplate() != null &&
+                deployment.getSpec().getTemplate().getSpec() != null &&
+                deployment.getSpec().getTemplate().getSpec().getContainers() != null &&
+                deployment.getSpec().getTemplate().getSpec().getContainers().size() > 0) {
+
+            List<Container> containers = deployment.getSpec().getTemplate().getSpec().getContainers();
+            return containers.get(containers.size() - 1); // last one
```

Comment:
Thinking about your proposal.. So the reason why you would like to move the logic of health checks from enrichers to generators is because health checks are becoming a lower level concept, because they are going to be embedded in containers. And I agree.

First of all we should check if the new image format is backward compatible with old "runtimes". If it is, there should be no problems for adding support for health checks in d-m-p, since it should be aware of the "builder" version. The new health check string could have been implemented as a label and let the engine do the rest (just supposing, did not check). The best form of communication between generator and enrichers is letting enrichers inspect images, so they can create probes even for images with bundled health checks, also outside the range of fabric8. We should check how to do it.

The second problem is that docker health checks seem almost equivalent to kubernetes exec liveness probes. They don't have the readiness probe concept, and don't have a http get equivalent. The fabric8 labels should fill these gaps.

So, in order to proceed, given that only one standard enricher is expected to survive the natural selection, I would merge the two health check enrichers into eg. the fabric8 enricher project and fix the spring-boot issue that was the subject of this PR :)
(It is blocking some activities).

I'll start to work on a new model for health checks, tracking things in a specific issue. Wdyt?